istioctl/cmd/describe.go

```diff
@@ -236,7 +240,41 @@ func describe() *cobra.Command {
 	return describeCmd
 }
 
-func validatePort(port v1.ServicePort, pod *v1.Pod) []string {
+func getIstioVersion(cd *configdump.Wrapper) string {
+	bootstrapDump, err := cd.GetBootstrapConfigDump()
+	if err == nil {
+		bootstrapNode := bootstrapDump.GetBootstrap().Node
+		if bootstrapNode.Metadata == nil {
+			// Happens if there has been no dynamic config, e.g. on Pilot itself
+			return noVersionMetadata
+		}
+
+		return asMyGogoValue(bootstrapNode.Metadata).keyAsString("ISTIO_VERSION")
+	}
+
+	return "undetected"
+}
+
+func containerPortOptional(istioVersion string) bool {
+	if istioVersion == noVersionMetadata {
+		return true
+	}
+	return supportsProtocolDetection(istioVersion)
+}
+
+func supportsProtocolDetection(istioVersion string) bool {
```

Comment:
can you reuse functions like https://github.com/istio/istio/blob/2dc426df846f3c3f73393fac233e8785024787fb/pilot/pkg/networking/util/util.go#L270-L273 and https://github.com/istio/istio/blob/c9ba34bf2aa8a9129a7a5c18124d301bd6a1f2c6/pilot/pkg/model/context.go#L396-L419?